clustered/server/src/main/java/org/ehcache/clustered/server/state/MessageTracker.java

```diff
@@ -80,6 +85,17 @@ public boolean isEmpty() {
    * Remove the contiguous seen msgIds from the nonContiguousMsgIds and update highestContiguousMsgId
    */
   private void reconcile() {
+
+    // This happens when a passive is started after Active has moved on and
+    // passive starts to see msgIDs starting from a number > 0.
+    if (highestContiguousMsgId == -1L && nonContiguousMsgIds.size() > 100) {
+      Long min = Collections.min(nonContiguousMsgIds);
+      LOGGER.info("Setting highestContiguousMsgId to {} from -1", min);
+      highestContiguousMsgId = min;
+    }
+
+    nonContiguousMsgIds.removeIf(x -> x <= highestContiguousMsgId);
```

Comment:
No its needed, since the point when `highestContiguousMsgId ` is first set, nonContiguousMsgIds might not have seen those msgIDs less than highestContiguousMsgId. Thus its needs to be cleaned up.